crates/typst/src/foundations/decimal.rs

```diff
@@ -12,63 +12,77 @@ use crate::World;
 /// constructing a decimal from a [floating-point number]($float), while
 /// supported, **is an imprecise conversion and therefore discouraged.** A
-/// warning will be raised if Typst detects that there was an accidental
-/// `{float}` to `{decimal}` cast through its constructor (e.g. if writing
-/// `{decimal(3.14)}` - note the lack of double quotes, indicating this is
-/// an accidental `{float}` cast and therefore imprecise). The precision of a
-/// `{float}` to `{decimal}` cast can be slightly improved by rounding the
-/// result to 15 digits with [`calc.round`]($calc.round), but there are still
-/// no precision guarantees for that kind of conversion.
+/// warning will be raised if Typst detects that there was an accidental `float`
+/// to `decimal` cast through its constructor (e.g. if writing `{decimal(3.14)}`
+/// - note the lack of double quotes, indicating this is an accidental `float`
+/// cast and therefore imprecise).
 ///
-/// In order to guard against accidental loss of precision, built-in operations
-/// between `{float}` and `{decimal}` are not supported and will raise an
-/// error. Certain `calc` functions, such as trigonometric functions and power
-/// between two real numbers, are also only supported for `{float}` (although
-/// raising `{decimal}` to integer exponents is supported). You can opt into
-/// potentially imprecise operations with the `{float(decimal)}` constructor,
-/// which casts the `{decimal}` number into a `{float}`, allowing for
-/// operations without precision guarantees.
+/// The precision of a `float` to `decimal` cast can be slightly improved by
+/// rounding the result to 15 digits with [`calc.round`]($calc.round), but there
+/// are still no precision guarantees for that kind of conversion.
 ///
-/// # Example
-/// ```example
-/// #decimal("3.14159265358979323846264338") \
-/// #(decimal("0.000000000000000000001") + decimal("0.000000000000000000002"))
-/// #(decimal("0.00002") * decimal("49.25652565")) \
-/// #(decimal("1") / 2048)
-/// ```
+/// # Operations
+/// Basic arithmetic operations are supported on two decimals and on pairs of
+/// decimals and integers.
+///
+/// Built-in operations between `float` and `decimal` are not supported in order
+/// to guard against accidental loss of precision. They will raise an error
+/// instead.
+///
+/// Certain `calc` functions, such as trigonometric functions and power between
+/// two real numbers, are also only supported for `float` (although raising
+/// `decimal` to integer exponents is supported). You can opt into potentially
+/// imprecise operations with the `{float(decimal)}` constructor, which casts
+/// the `decimal` number into a `float`, allowing for operations without
+/// precision guarantees.
+///
+/// # Displaying decimals
+/// To display a decimal, simply insert the value into the document. To only
+/// display a certain number of digits, [round]($calc.round) the decimal first.
```

Comment:
Could also mention `str(the decimal)` to obtain it as a string perhaps? (Not 100% necessary, but a thought I had while reading this.)